internal/blocksync/reactor.go

```diff
@@ -549,12 +545,13 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 
 			// see if there are any blocks to sync
 			first, second, extCommit := r.pool.PeekTwoBlocks()
-			if first == nil || second == nil || extCommit == nil {
-				if first != nil && extCommit == nil {
-					// See https://github.com/tendermint/tendermint/pull/8433#discussion_r866790631
-					panic(fmt.Errorf("peeked first block without extended commit at height %d - possible node store corruption", first.Height))
-				}
-				// we need all to sync the first block
+			if state.ConsensusParams.ABCI.VoteExtensionsEnabled(state.LastBlockHeight+1) &&
+				first != nil && extCommit == nil {
```

Comment:
Would we need to rearrange this condition to call `EnsureExtensions`?
Or do we have the guarantee that there'll always be extensions since we've modified the logic of `respondToPeer`? (or maybe we need to do `EnsureExtensions` also in `respondToPeer`)

Just asking (not sure myself)

Reply:
Forget this, I see we do it in 592